piker/brokers/kraken.py

```diff
@@ -523,50 +538,65 @@ async def handle_order_requests(
             # validate
             temp_id = next(userref_counter)
             order = BrokerdOrder(**request_msg)
+            def slashinsert(str):
+                midPoint = len(str)//2
+                return str[:midPoint] + '/' + str[midPoint:]
+            # Send order via websocket
+            order_msg = {
+                          "event": "addOrder",
+                          "ordertype": "limit",
+                          "pair": slashinsert(order.symbol.upper()),
+                          "price": str(order.price),
+                          "token": token,
+                          "type": order.action,
+                          "volume": str(order.size)
+                        }
+                    
+
+            await ws.send_msg(order_msg)
 
             # call our client api to submit the order
-            resp = await client.submit_limit(
-
-                oid=order.oid,
-                symbol=order.symbol,
-                price=order.price,
-                action=order.action,
-                size=order.size,
-                ## XXX: how do I handle new orders
-                reqid=temp_id,
-            )
-
-            err = resp['error']
-            if err:
-                log.error(f'Failed to submit order')
-                await ems_order_stream.send(
-                    BrokerdError(
-                        oid=order.oid,
-                        reqid=temp_id,
-                        symbol=order.symbol,
-                        reason="Failed order submission",
-                        broker_details=resp
-                    ).dict()
-                )
-            else:
-                ## TODO: handle multiple cancels
-                ##       txid is an array of strings
-                reqid = resp['result']['txid'][0]
-                # deliver ack that order has been submitted to broker routing
-                await ems_order_stream.send(
-                    BrokerdOrderAck(
-
-                        # ems order request id
-                        oid=order.oid,
-
-                        # broker specific request id
-                        reqid=reqid,
-
-                        # account the made the order
-                        account=order.account
-
-                    ).dict()
-                )
+            #resp = await client.submit_limit(
```

Comment:
Yeah, i don't this you have to necessarily switch to the `ws` version here if the http api worked well already (which it did when i tested it), but hey great if we want both and or the ws one does work better.

brining in the ws support could also come in a follow up issue + PR 😺

Reply:
It would be pretty cool to test the latency different between these.
I got a feeling we might also not run into as many rate limiting problems on the `kraken` side with ws, but it's probably worth testing.

Reply:
performance aside, using the http api solves the issue of having to keep track of mappings, which I think becomes tricky if you have existing orders that reference the same `userref`.

That being said, the `ws` exercise was good and the code can easily be repurposed for the `ownTrades` stream to keep track of successful trades.

Thoughts? @goodboy @guilledk